### PR TITLE
Enable customizing the default visa timeout

### PR DESCRIPTION
**Pull request by nfelt14**
## Proposed changes

This enables customizing the default VISA timeout via the configuration file or environment variable.

Addresses #294 

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] New feature (non-breaking change which adds functionality)
- [x] Functionality update (non-breaking change which updates or changes existing functionality)
- [x] Documentation update (an update to enhance the user experience when reading through the docs)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have followed the guidelines in the [CONTRIBUTING](https://github.com/tektronix/tm_devices/blob/main/CONTRIBUTING.md) document
- [x] I have signed the CLA
- [x] I have checked to ensure there aren't other open [Pull Requests](https://github.com/tektronix/tm_devices/pulls) for the same update/change
- [x] I have created (or updated) an [Issue](https://github.com/tektronix/tm_devices/issues) to track the status of this update/change and updated the link in this PR description (see above in the **Proposed changes** section) using the wording `Addresses #<issue_number>`
- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] Basic linting passes locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added necessary documentation (if appropriate)
- [x] I have updated the Changelog with a brief description of my changes

### CODE REVIEW

File: src/tm_devices/components/dm_config_parser.py [hunk 1]
```diff
@@ -353,15 +352,22 @@ def __parse_env_options(self) -> DMConfigOptions:
...
+            for arg in os.getenv(self.OPTIONS_ENV_VARIABLE, "").split(",")
+            if arg
+        }
         valid_config_options = {option.upper() for option in get_type_hints(DMConfigOptions)}
         if valid_config_options.issuperset(options_list):
             options = {
-                arg_name.lower(): arg_name in options_list for arg_name in valid_config_options
+                arg_name.lower(): _convert_config_values(options_list[arg_name])
+                if arg_name in options_list
+                else False
```
Comment (nfelt14) [inline src/tm_devices/components/dm_config_parser.py:367]: Should this be `None` if nothing is found?

File: src/tm_devices/drivers/device.py [hunk 2]
```diff
@@ -262,7 +262,7 @@ def name(self) -> str:
 
         Usually something like "SCOPE 1"
         """
-        return f"{self.device_type} {self.device_number}"
+        return f"{self.device_type}" + (f" {self.device_number}" if self.device_number > 0 else "")
```
Comment (nfelt14) [inline src/tm_devices/drivers/device.py:265]: This just makes the printout look nicer, no more `SCOPE -1` in the stdout.

**Comment by nfelt14** [comment]
Currently blocked by https://github.com/actions/setup-python/issues/935

File: src/tm_devices/components/dm_config_parser.py (hunk 1 above)
Comment (michaelwagoner) [reply src/tm_devices/components/dm_config_parser.py:367]: That would make sense to me

**Review by michaelwagoner — DISMISSED** [review]

File: src/tm_devices/drivers/device.py (hunk 2 above)
Comment (ldantek) [inline src/tm_devices/drivers/device.py:265]:
```suggestion
        return f"{self.device_type} {self.device_number or  ''}".strip()
```

File: src/tm_devices/components/dm_config_parser.py [hunk 3]
```diff
@@ -353,15 +352,22 @@ def __parse_env_options(self) -> DMConfigOptions:
...
+            else arg.split("=", maxsplit=1)[-1]
+            for arg in os.getenv(self.OPTIONS_ENV_VARIABLE, "").split(",")
+            if arg
+        }
         valid_config_options = {option.upper() for option in get_type_hints(DMConfigOptions)}
         if valid_config_options.issuperset(options_list):
             options = {
-                arg_name.lower(): arg_name in options_list for arg_name in valid_config_options
+                arg_name.lower(): _convert_config_values(options_list[arg_name])
+                if arg_name in options_list
```
Comment (ldantek) [inline src/tm_devices/components/dm_config_parser.py:366]: this is already handled by the issuperset if statement

**Review by ldantek — APPROVED** [review]

**Pull request merged**